File: newspark-art/TypeScript/tests/pages/TasksPage.ts
import { Page, Locator } from '@playwright/test';

export class TasksPage {
  private page: Page;
  private nextButton: Locator;
  private previousButton: Locator;
  private currentPageNumber: Locator;
  private totalPages: Locator;
  private titleInput: Locator;
  private editTitleInput: Locator;
  private addTaskButton: Locator;
  private applyButton: Locator;
  private confirmDeleteButton: Locator;
  private url: string;

  constructor(page: Page) {
    this.page = page;
    this.nextButton = page.locator('[data-testid="next-button"]');
    this.previousButton = page.locator('[data-testid="previous-button"]');
    this.currentPageNumber = page.locator('[data-testid="current-page-number"]');
    this.totalPages = page.locator('[data-testid="total-page-number"]');
    this.titleInput = page.locator('[data-testid="title-input"]');
    this.editTitleInput = page.locator('[data-testid="edit-title-input"]');
    this.addTaskButton = page.locator('[data-testid="add-task-button"]');
    this.applyButton = page.locator('[data-testid="apply-button"]');
    this.confirmDeleteButton = page.locator('[data-testid="delete-button"]');
    this.url = "http://localhost:5173/tasks";
  }

  async navigate(): Promise<void> {
    await this.page.goto(this.url);
  }

  async clickNext(): Promise<void> {
    await this.nextButton.click();
  }

  async clickPrevious(): Promise<void> {
    await this.previousButton.click();
  }

  async getCurrentPageNumber(): Promise<string | null> {
    return await this.currentPageNumber.textContent();
  }

  async getTotalPages(): Promise<string | null> {
    return await this.totalPages.textContent();
  }

  async isNextButtonVisible(): Promise<boolean> {
    return await this.nextButton.isVisible();
  }

  async isPreviousButtonVisible(): Promise<boolean> {
    return await this.previousButton.isVisible();
  }

  async isPaginationVisible(): Promise<boolean> {
    return (await this.isNextButtonVisible()) && (await this.isPreviousButtonVisible());
  }

  async isNextButtonEnabled(): Promise<boolean> {
    return await this.nextButton.isVisible();
  }

  async isPreviousButtonEnabled(): Promise<boolean> {
    return await this.previousButton.isVisible();
  }

  async isNextButtonDisabled(): Promise<boolean> {
    return (await this.nextButton.getAttribute('disabled')) !== null;
  }

  async isPreviousButtonDisabled(): Promise<boolean> {
    return (await this.previousButton.getAttribute('disabled')) !== null;
  }

  async enterTaskTitle(taskTitle: string): Promise<void> {
    await this.titleInput.fill(taskTitle);
  }

  async clickAddTaskButton(): Promise<void> {
    await this.addTaskButton.click();
  }

  async isTaskInTable(taskTitle: string): Promise<boolean> {
    return await this.page.locator(`table tr:has-text("${taskTitle}")`).isVisible({ timeout: 3000 });
  }

  async clickEditButton(taskTitle: string): Promise<void> {
    await this.page.locator(`table tr:has-text("${taskTitle}") >> [data-testid*="edit-task-button"]`).click();
  }

  async updateTaskTitle(newTaskTitle: string): Promise<void> {
    await this.editTitleInput.fill(newTaskTitle);
  }

  async clickApplyButton(): Promise<void> {
    await this.applyButton.click();
  }

  async clickDeleteButton(taskTitle: string): Promise<void> {
    await this.page.locator(`table tr:has-text("${taskTitle}") >> [data-testid*="delete-task-button"]`).click();
  }

  async confirmDeletion(): Promise<void> {
    await this.confirmDeleteButton.click();
  }

  async clearTaskInput(): Promise<void> {
    await this.titleInput.fill('');
  }

  async clearEditInput(): Promise<void> {
    await this.editTitleInput.fill('');
  }
}

File: newspark-art/Python/pages/tasks_page.py
from playwright.sync_api import Page

class TasksPage:
    def __init__(self, page: Page):
        self.page = page
        self.next_button = page.locator('[data-testid="next-button"]')
        self.previous_button = page.locator('[data-testid="previous-button"]')
        self.current_page_number = page.locator('[data-testid="current-page-number"]')
        self.total_pages = page.locator('[data-testid="total-page-number"]')
        self.title_input = page.locator('[data-testid="title-input"]')
        self.edit_title_input = page.locator('[data-testid="edit-title-input"]')
        self.add_task_button = page.locator('[data-testid="add-task-button"]')
        self.apply_button = page.locator('[data-testid="apply-button"]')
        self.confirm_delete_button = page.locator('[data-testid="delete-button"]')
        self.url = "http://localhost:5173/tasks"

    def navigate(self):
        self.page.goto(self.url)

    def click_next(self):
        self.next_button.click()

    def click_previous(self):
        self.previous_button.click()

    def get_current_page_number(self):
        return self.current_page_number.text_content()

    def get_total_pages(self):
        return self.total_pages.text_content()

    def is_next_button_visible(self):
        return self.next_button.is_visible()

    def is_previous_button_visible(self):
        return self.previous_button.is_visible()

    def is_pagination_visible(self):
        return self.is_next_button_visible() and self.is_previous_button_visible()

    def is_next_button_enabled(self):
        return self.next_button.is_visible()

    def is_previous_button_enabled(self):
        return self.previous_button.is_visible()

    def is_next_button_disabled(self):
        return self.next_button.get_attribute('disabled') is not None

    def is_previous_button_disabled(self):
        return self.previous_button.get_attribute('disabled') is not None

    def enter_task_title(self, task_title: str):
        self.title_input.fill(task_title)

    def click_add_task_button(self):
        self.add_task_button.click()

    def is_task_in_table(self, task_title: str):
        return self.page.locator(f'table tr:has-text("{task_title}")').is_visible(timeout=3000)

    def click_edit_button(self, task_title: str):
        self.page.locator(f'table tr:has-text("{task_title}") >> [data-testid*="edit-task-button"]').click()

    def update_task_title(self, new_task_title: str):
        self.edit_title_input.fill(new_task_title)

    def click_apply_button(self):
        self.apply_button.click()

    def click_delete_button(self, task_title: str):
        self.page.locator(f'table tr:has-text("{task_title}") >> [data-testid*="delete-task-button"]').click()

    def confirm_deletion(self):
        self.confirm_delete_button.click()

    def clear_task_input(self):
        self.title_input.fill('')

    def clear_edit_input(self):
        self.edit_title_input.fill('')
